civicrm.user.inc

```diff
@@ -11,6 +11,18 @@ use Drupal\Core\Session\AccountInterface;
  * Implements hook_user_login().
  */
 function civicrm_user_login(AccountInterface $account) {
+  $civicrm = \Drupal::service('civicrm');
+  $civicrm->initialize();
+
+  if (\Drupal::hasService('masquerade')) {
+    // In theory this is harmless whether we masquerade or not, but for now narrowing
+    // the scope, in case it causes regressions.
+    // Also, there is a isMasquerading() function, but it does not seem to be
+    // initialized at this point (it always returns false).
+    $session = \CRM_Core_Session::singleton();
+    $session->reset(FALSE);
+  }
+
   \Drupal::service('civicrm')->synchronizeUser($account);
 }
```

Comment:
just to reduce LOC 
```php
function civicrm_user_login(AccountInterface $account) {
  $civicrm = \Drupal::service('civicrm');
  $civicrm->initialize();

  if (\Drupal::hasService('masquerade')) {
    // In theory this is harmless whether we masquerade or not, but for now narrowing
    // the scope, in case it causes regressions.
    // Also, there is a isMasquerading() function, but it does not seem to be
    // initialized at this point (it always returns false).
    $session = \CRM_Core_Session::singleton()->reset(FALSE);
  }
  $civicrm->synchronizeUser($account);
```

Reply:
:+1: fixed

Reply:
thanks :)

Reply:
It works!